src/ConfigManager.ts

```diff
@@ -37,6 +68,13 @@ export default class ConfigManager {
       logger.error({ invalidProperty: 'proxyHosts' }, 'config is invalid, missing property');
     } else {
       this.loadProxyHosts(config.proxyHosts);
+      if (config.proxyListeningPort) {
+        const prevPort = this.proxyListeningPort;
+        this.proxyListeningPort = ConfigManager.parsePort(config.proxyListeningPort);
```

Comment:
Should we be updating the local variable with a value, which we might not use in the end? (Because of the `if` on the next line?)
And what happens when the new value from the config returns a null?

Reply:
Yes because the if on the next line is so the event is not emitted on startup. The proxyListeningPort still needs to be set on startup. This prevents the program from setting up the proxy twice when the application starts. The only case where prevPort will be null is on startup and we do not need to emit an event if the previous port is the same as the current port on a config change. 

If parsePort returns a null here then the env port will be used or port 80 if there is no env port. The if statements in the getProxyListeningPort function will see a null as false.

Reply:
Ah perfect, forgot that the `getProxyListeningPort` function also performs some checks 👍